### PR TITLE
Move TCNo and INN from calculator to localized providers

### LINKED ISSUE #106: Move Russian Inn and Turkish TCNo from Calculator to localized providers
[`Calculator/Inn.php`](https://github.com/FakerPHP/Faker/blob/main/src/Faker/Calculator/Inn.php) verifies the Russian tax identification number (INN number), [`Calculator/TCNo.php`](https://github.com/FakerPHP/Faker/blob/main/src/Faker/Calculator/TCNo.php) the Turkish Identity Number - sounds like stuff that should be in localized providers?

Should we move the `Calculator/Inn.php` methods to `ru_RU/Company.php`, mark them deprecated in `Inn` and call the new methods in `Company.php` for BC? Similarly with `TCNo`/`tr_TR/Person.php`.

### PR DESCRIPTION
**Pull request by krsriq**
### What is the reason for this PR?

`Inn.php` verifies the Russian tax identification number (INN number), `TCNo.php` the Turkish Identity Number, but both are currently placed in the core `Calculator`.

- [ ] A new feature
- [x] Fixed an issue (resolve #106)

### Author's checklist

- [x] Follow the [Contribution Guide](https://github.com/FakerPHP/Faker/blob/main/.github/CONTRIBUTING.md)
- [ ] New features and changes are [documented](https://github.com/FakerPHP/fakerphp.github.io)

### Summary of changes

- Moved `Calculator\Inn` methods to  `Provider\ru_RU\Company`, `Calculator\TCNo` methods to `Provider\tr_TR\Person`
- Marked `Inn` and `TcNo` classes and methods deprecated
- Called the new `Company`/`Person` methods from `Inn` and `TcNo` for BC
- Moved the test methods from `Calculator` to their respective `Company` and `Person` test classes

### Review checklist

- [ ] All checks have passed
- [ ] Changes are approved by maintainer

### CODE REVIEW

File: src/Faker/Calculator/Inn.php [hunk 1]
```diff
@@ -2,36 +2,36 @@
...
 class Inn
 {
     /**
      * Generates INN Checksum
-     *
      * https://ru.wikipedia.org/wiki/%D0%98%D0%B4%D0%B5%D0%BD%D1%82%D0%B8%D1%84%D0%B8%D0%BA%D0%B0%D1%86%D0%B8%D0%BE%D0%BD%D0%BD%D1%8B%D0%B9_%D0%BD%D0%BE%D0%BC%D0%B5%D1%80_%D0%BD%D0%B0%D0%BB%D0%BE%D0%B3%D0%BE%D0%BF%D0%BB%D0%B0%D1%82%D0%B5%D0%BB%D1%8C%D1%89%D0%B8%D0%BA%D0%B0
      *
      * @param string $inn
      * @return string Checksum (one digit)
+     *@deprecated Use {@link \Faker\Provider\ru_RU\Company::inn10Checksum()} instead
```
Comment (bramceulemans) [inline src/Faker/Calculator/Inn.php:17]: You forgot a space here, before `@deprecated`
```suggestion
     * @deprecated use {@link \Faker\Provider\ru_RU\Company::inn10Checksum()} instead
```

File: src/Faker/Calculator/TCNo.php [hunk 2]
```diff
@@ -12,41 +14,28 @@ class TCNo
...
-        $eleventhDigit = ($evenSum + $oddSum + $tenthDigit) % 10;
-
-        return $tenthDigit . $eleventhDigit;
+        return \Faker\Provider\tr_TR\Person::tcNoChecksum($identityPrefix);
     }
 
     /**
      * Checks whether a TCNo has a valid checksum
      *
+     * @deprecated Use {@link \Faker\Provider\tr_TR\Person::tcNoIsValid()} instead
```
Comment (bramceulemans) [inline src/Faker/Calculator/TCNo.php:31]: Please be consistent in where you place the `@deprecated` (after `@return`)

**Review by bramceulemans — CHANGES_REQUESTED** [review]

File: src/Faker/Calculator/TCNo.php (hunk 2 above)
Comment (krsriq) [reply src/Faker/Calculator/TCNo.php]: Looks better now, thanks.

File: src/Faker/Calculator/Inn.php [hunk 3]
```diff
@@ -2,36 +2,36 @@
 
 namespace Faker\Calculator;
 
+/**
+ * @deprecated Moved to ru_RU\Company, use {@link \Faker\Provider\ru_RU\Company}.
```
Comment (bramceulemans) [inline src/Faker/Calculator/Inn.php:6]:
```suggestion
 * @deprecated moved to ru_RU\Company, use {@link \Faker\Provider\ru_RU\Company}.
```

File: src/Faker/Calculator/Inn.php [hunk 4]
```diff
@@ -2,36 +2,36 @@
...
-        return (string) (($sum % 11) % 10);
+        return \Faker\Provider\ru_RU\Company::inn10Checksum($inn);
     }
 
     /**
      * Checks whether an INN has a valid checksum
      *
      * @param string $inn
      * @return bool
+     * @deprecated Use {@link \Faker\Provider\ru_RU\Company::inn10IsValid()} instead
```
Comment (bramceulemans) [inline src/Faker/Calculator/Inn.php:30]:
```suggestion
     * @deprecated use {@link \Faker\Provider\ru_RU\Company::inn10IsValid()} instead
```

File: src/Faker/Calculator/TCNo.php [hunk 5]
```diff
@@ -2,8 +2,10 @@
 
 namespace Faker\Calculator;
 
-use InvalidArgumentException;
-
+/**
+ * @deprecated Moved to tr_TR\Person, use {@link \Faker\Provider\tr_TR\Person}.
```
Comment (bramceulemans) [inline src/Faker/Calculator/TCNo.php:6]:
```suggestion
 * @deprecated moved to tr_TR\Person, use {@link \Faker\Provider\tr_TR\Person}.
```

File: src/Faker/Calculator/TCNo.php [hunk 6]
```diff
@@ -14,39 +16,24 @@ class TCNo
      *
      * @param string $identityPrefix
      * @return string Checksum (two digit)
+     * @deprecated Use {@link \Faker\Provider\tr_TR\Person::tcNoChecksum()} instead
```
Comment (bramceulemans) [inline src/Faker/Calculator/TCNo.php:19]:
```suggestion
     * @deprecated use {@link \Faker\Provider\tr_TR\Person::tcNoChecksum()} instead
```

File: src/Faker/Calculator/TCNo.php [hunk 7]
```diff
@@ -14,39 +16,24 @@ class TCNo
...
-        return $tenthDigit . $eleventhDigit;
+        return \Faker\Provider\tr_TR\Person::tcNoChecksum($identityPrefix);
     }
 
     /**
      * Checks whether a TCNo has a valid checksum
      *
      * @param string $tcNo
      * @return bool
+     * @deprecated Use {@link \Faker\Provider\tr_TR\Person::tcNoIsValid()} instead
```
Comment (bramceulemans) [inline src/Faker/Calculator/TCNo.php:32]:
```suggestion
     * @deprecated use {@link \Faker\Provider\tr_TR\Person::tcNoIsValid()} instead
```

**Review by bramceulemans — CHANGES_REQUESTED** [review]

File: src/Faker/Calculator/TCNo.php (hunk 7 above)
Comment (krsriq) [reply src/Faker/Calculator/TCNo.php:32]: Okay, then we probably want to change the ones in 

https://github.com/FakerPHP/Faker/blob/2e1bc4c64cdf8cb101cc416b4a660129af6029f5/src/Faker/Provider/Barcode.php#L39
https://github.com/FakerPHP/Faker/blob/2e1bc4c64cdf8cb101cc416b4a660129af6029f5/src/Faker/Provider/Barcode.php#L24
https://github.com/FakerPHP/Faker/blob/2e1bc4c64cdf8cb101cc416b4a660129af6029f5/src/Faker/Provider/Image.php#L15

etc. also ;)

File: src/Faker/Provider/ru_RU/Company.php [hunk 8]
```diff
@@ -96,7 +96,26 @@ public static function companyNameSuffix()
...
+        return self::inn10($area_code);
+    }
+
+    /**
+     * Generates a Russian Taxpayer Personal Identification Number
+     *
+     * @param  string  $area_code
+     * @return string
+     */
+    public static function inn10($area_code = "")
```
Comment (bramceulemans) [inline src/Faker/Provider/ru_RU/Company.php:118]:
```suggestion
    public static function inn10($area_code = '')
```

**Review by bramceulemans — CHANGES_REQUESTED** [review]
One small change requested, otherwise it looks good!

File: src/Faker/Provider/ru_RU/Company.php [hunk 9]
```diff
@@ -96,7 +96,26 @@ public static function companyNameSuffix()
...
 
+    /**
+     * Generates a Russian Taxpayer Personal Identification Number
+     *
+     * @param  string  $area_code
+     * @return string
+     * @deprecated use {@link \Faker\Provider\ru_RU\Company::inn10()} instead
+     * @see \Faker\Provider\ru_RU\Company::inn10()
+     */
     public static function inn($area_code = "")
```
Comment (bramceulemans) [inline src/Faker/Provider/ru_RU/Company.php:107]:
```suggestion
    public static function inn($area_code = '')
```

**Review by bramceulemans — CHANGES_REQUESTED** [review]

File: src/Faker/Provider/ru_RU/Company.php [hunk 10]
```diff
@@ -106,14 +125,44 @@ public static function inn($area_code = "")
         }
         $area_code = str_pad($area_code, 2, '0', STR_PAD_LEFT);
         $inn_base =  $area_code . static::numerify('#######');
-        return $inn_base . \Faker\Calculator\Inn::checksum($inn_base);
+        return $inn_base . self::inn10Checksum($inn_base);
     }
 
     public static function kpp($inn = "")
```
Comment (bramceulemans) [inline src/Faker/Provider/ru_RU/Company.php:131]:
```suggestion
    public static function kpp($inn = '')
```

File: src/Faker/Provider/ru_RU/Company.php [hunk 11]
```diff
@@ -106,14 +125,44 @@ public static function inn($area_code = "")
...
     {
         if ($inn == "" || strlen($inn) < 4) {
```
Comment (bramceulemans) [inline src/Faker/Provider/ru_RU/Company.php:133]:
```suggestion
        if ($inn === '' || strlen($inn) < 4) {
```

File: src/Faker/Provider/ru_RU/Company.php [hunk 12]
```diff
@@ -106,14 +125,44 @@ public static function inn($area_code = "")
...
-            $inn = static::inn();
+            $inn = self::inn10();
         }
         return substr($inn, 0, 4) . "01001";
```
Comment (bramceulemans) [inline src/Faker/Provider/ru_RU/Company.php:136]:
```suggestion
        return substr($inn, 0, 4) . '01001';
```

File: src/Faker/Provider/ru_RU/Company.php [hunk 13]
```diff
@@ -96,7 +96,26 @@ public static function companyNameSuffix()
...
+
+    /**
+     * Generates a Russian Taxpayer Personal Identification Number
+     *
+     * @param  string  $area_code
+     * @return string
+     */
+    public static function inn10($area_code = '')
     {
         if ($area_code === "" || intval($area_code) == 0) {
```
Comment (bramceulemans) [inline src/Faker/Provider/ru_RU/Company.php:120]:
```suggestion
        if ($area_code === '' || intval($area_code) === 0) {
```

File: src/Faker/Provider/tr_TR/Person.php [hunk 14]
```diff
@@ -105,8 +105,51 @@ public static function titleFemale()
...
+        if (strlen((string) $identityPrefix) !== 9) {
+            throw new InvalidArgumentException('Argument should be an integer and should be 9 digits.');
+        }
+
+        $oddSum = 0;
+        $evenSum = 0;
+
+        $identityArray = array_map('intval', str_split($identityPrefix)); // Creates array from int
+        foreach ($identityArray as $index => $digit) {
+            if ($index % 2 == 0) {
```
Comment (bramceulemans) [inline src/Faker/Provider/tr_TR/Person.php:132]:
```suggestion
            if ($index % 2 === 0) {
```

**Review by bramceulemans — CHANGES_REQUESTED** [review]

File: src/Faker/Provider/ru_RU/Company.php (hunk 12 above)
Comment (krsriq) [reply src/Faker/Provider/ru_RU/Company.php:136]: How about we enable [`single_quote`](https://docs.styleci.io/fixers#single_quote) in StyleCI for these?

Comment (bramceulemans) [reply src/Faker/Provider/ru_RU/Company.php]: Couldn't agree more.

**Review by bramceulemans — APPROVED** [review]

**Comment by bramceulemans** [comment]
Let's enable single_quotes and then rebase.

**Comment by krsriq** [comment]
> Let's enable `single_quotes` and then rebase.

@bramceulemans - rebased a while ago.

File: src/Faker/Calculator/Inn.php [hunk 15]
```diff
@@ -2,36 +2,36 @@
...
 
+/**
+ * @deprecated moved to ru_RU\Company, use {@link \Faker\Provider\ru_RU\Company}.
+ * @see \Faker\Provider\ru_RU\Company
+ */
 class Inn
 {
     /**
      * Generates INN Checksum
-     *
```
Comment (GrahamCampbell) [inline src/Faker/Calculator/Inn.php:9]: ?

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Faker/Calculator/Inn.php (hunk 15 above)
Comment (krsriq) [reply src/Faker/Calculator/Inn.php]: Re-added that empty line.

**Comment by bramceulemans** [comment]
Looks good to me now.
/cc @GrahamCampbell

File: test/Faker/Provider/tr_TR/PersonTest.php [hunk 16]
```diff
@@ -14,10 +13,55 @@ public function testTCNo()
...
+            ['19874561012', true],
+
+            ['11111111111', false],
+            ['11234567899', false],
+        ];
+    }
+
+    /**
+     * @dataProvider tcNoValidatorProvider
+     * @param $tcNo
```
Comment (GrahamCampbell) [inline test/Faker/Provider/tr_TR/PersonTest.php:57]: invalid syntax

File: test/Faker/Provider/tr_TR/PersonTest.php [hunk 17]
```diff
@@ -14,10 +13,55 @@ public function testTCNo()
...
+     * @param $isValid
```
Comment (GrahamCampbell) [inline test/Faker/Provider/tr_TR/PersonTest.php:58]: same

File: test/Faker/Provider/tr_TR/PersonTest.php [hunk 18]
```diff
@@ -14,10 +13,55 @@ public function testTCNo()
...
+            ['350630743', '78'],
+            ['550600932', '88'],
+            ['487932947', '70'],
+            ['168113862', '40']
+        ];
+    }
+
+    /**
+     * @dataProvider tcNoChecksumProvider
+     * @param $tcNo
```
Comment (GrahamCampbell) [inline test/Faker/Provider/tr_TR/PersonTest.php:33]: same

File: test/Faker/Provider/tr_TR/PersonTest.php [hunk 19]
```diff
@@ -14,10 +13,55 @@ public function testTCNo()
...
+     * @param $checksum
```
Comment (GrahamCampbell) [inline test/Faker/Provider/tr_TR/PersonTest.php:34]: same

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: test/Faker/Provider/tr_TR/PersonTest.php (hunk 16 above)
Comment (krsriq) [reply test/Faker/Provider/tr_TR/PersonTest.php]: Added param type now.

File: test/Faker/Provider/tr_TR/PersonTest.php (hunk 17 above)
Comment (krsriq) [reply test/Faker/Provider/tr_TR/PersonTest.php]: Added param type now.

File: test/Faker/Provider/tr_TR/PersonTest.php (hunk 18 above)
Comment (krsriq) [reply test/Faker/Provider/tr_TR/PersonTest.php]: Added param type now.

**Comment by krsriq** [comment]
@GrahamCampbell added `@dataProvider` `@param` types here now (and fixed them in the other class that was missing this with #147).

File: src/Faker/Calculator/Inn.php [hunk 20]
```diff
@@ -11,27 +15,24 @@ class Inn
...
     {
-        $multipliers = [2, 4, 10, 3, 5, 9, 4, 6, 8];
-        $sum = 0;
-
-        for ($i = 0; $i < 9; $i++) {
-            $sum += (int) $inn[$i] * $multipliers[$i];
-        }
-
-        return (string) (($sum % 11) % 10);
+        return \Faker\Provider\ru_RU\Company::inn10Checksum($inn);
```
Comment (pimjansen) [inline src/Faker/Calculator/Inn.php:23]: Im not happy with this where we have a generic method call a locale specific?

File: src/Faker/Calculator/Inn.php [hunk 21]
```diff
@@ -11,27 +15,24 @@ class Inn
...
      *
      * @param string $inn
      * @return bool
+     * @deprecated use {@link \Faker\Provider\ru_RU\Company::inn10IsValid()} instead
+     * @see \Faker\Provider\ru_RU\Company::inn10IsValid()
      */
     public static function isValid($inn)
     {
-        return strlen($inn) === 10 && self::checksum($inn) === $inn[9];
+        return \Faker\Provider\ru_RU\Company::inn10IsValid($inn);
```
Comment (pimjansen) [inline src/Faker/Calculator/Inn.php:36]: Same here

File: src/Faker/Calculator/TCNo.php [hunk 22]
```diff
@@ -14,39 +16,24 @@ class TCNo
...
-            } else {
-                $oddSum += $digit;
-            }
-        }
-
-        $tenthDigit = (7 * $evenSum - $oddSum) % 10;
-        $eleventhDigit = ($evenSum + $oddSum + $tenthDigit) % 10;
-
-        return $tenthDigit . $eleventhDigit;
+        return \Faker\Provider\tr_TR\Person::tcNoChecksum($identityPrefix);
```
Comment (pimjansen) [inline src/Faker/Calculator/TCNo.php:24]: Same here

File: src/Faker/Calculator/TCNo.php [hunk 23]
```diff
@@ -14,39 +16,24 @@ class TCNo
...
      *
      * @param string $tcNo
      * @return bool
+     * @deprecated use {@link \Faker\Provider\tr_TR\Person::tcNoIsValid()} instead
+     * @see \Faker\Provider\tr_TR\Person::tcNoIsValid()
      */
     public static function isValid($tcNo)
     {
-        return self::checksum(substr($tcNo, 0, -2)) === substr($tcNo, -2, 2);
+        return \Faker\Provider\tr_TR\Person::tcNoIsValid($tcNo);
```
Comment (pimjansen) [inline src/Faker/Calculator/TCNo.php:37]: Same

File: src/Faker/Calculator/Inn.php (hunk 21 above)
Comment (krsriq) [reply src/Faker/Calculator/Inn.php:36]: @pimjansen the problem here is that `INN` and `TCNo` _are_ locale specific but somehow ended up in the base `Calculator`. 

We keep them in `Calculator` (and call the locale specific providers) and mark them deprecated for BC , but they should be removed from the `Calculator` asap IMO - especially considering Faker 2.0, where it makes no sense to have this in the core.

**Comment by pimjansen** [comment]
@krsriq lgtm after a green build

**Comment by krsriq** [comment]
@pimjansen done!

**Pull request merged**